addOns/pscan/src/main/java/org/zaproxy/addon/pscan/internal/scanner/PassiveScanTask.java

```diff
@@ -0,0 +1,230 @@
+                            scanner.scanHttpRequestSend(msg, href.getHistoryId());
+                            scanned = true;
+                        } else {
+                            Stats.incCounter("stats.pscan.reqBodyTooBig");
+                            LOGGER.debug(
+                                    "Request to {} body size {} larger than max configured {}",
+                                    msg.getRequestHeader().getURI(),
+                                    msg.getRequestBody().length(),
+                                    maxBodySize);
+                        }
+                        if (msg.isResponseFromTargetHost()) {
+                            if (maxBodySize <= 0 || msg.getResponseBody().length() < maxBodySize) {
+                                scanner.scanHttpResponseReceive(msg, href.getHistoryId(), src);
+                                scanned = true;
+                            } else {
+                                Stats.incCounter("stats.pscan.respBodyTooBig");
+                                LOGGER.debug(
+                                        "Response from {} body size {} larger than max configured {}",
+                                        msg.getRequestHeader().getURI(),
+                                        msg.getResponseBody().length(),
+                                        maxBodySize);
+                            }
+                        }
+                        if (scanned) {
+                            long timeTaken = System.currentTimeMillis() - scanRuleStartTime;
+                            if (scanner instanceof PluginPassiveScanner) {
+                                PluginPassiveScanner pps = (PluginPassiveScanner) scanner;
+                                Stats.incCounter(
+                                        "stats.pscan." + pps.getPluginId() + ".time", timeTaken);
+                            }
+                            // TODO remove at some point
+                            Stats.incCounter("stats.pscan." + scanner.getName(), timeTaken);
+                            if (timeTaken > 5000) {
+                                // Took over 5 seconds, thats not ideal
+                                String responseInfo = "";
+                                if (msg.isResponseFromTargetHost()) {
+                                    responseInfo =
+                                            msg.getResponseHeader()
+                                                            .getHeader(HttpHeader.CONTENT_TYPE)
+                                                    + " "
+                                                    + msg.getResponseBody().length();
+                                }
+                                LOGGER.warn(
+                                        "Passive Scan rule {} took {} seconds to scan {} {}",
+                                        scanner.getName(),
+                                        timeTaken / 1000,
```

Comment:
Is ChronoUnit or TimeUnit to avoid the magic number?

Reply:
Updated.